src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/NamedPipeBasedRazorProjectInfoDriver.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.ProjectSystem;
+using Microsoft.AspNetCore.Razor.Utilities;
+using Microsoft.CodeAnalysis.Razor.Logging;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+
+internal class NamedPipeBasedRazorProjectInfoDriver : AbstractRazorProjectInfoDriver
+{
+    NamedPipeClientStream? _namedPipe;
+    Task? _readTask;
+
+    public NamedPipeBasedRazorProjectInfoDriver(ILoggerFactory loggerFactory) : base(loggerFactory, TimeSpan.FromMilliseconds(200))
+    {
+        StartInitialization();
+    }
+
+    public async Task ConnectAsync(string pipeName, CancellationToken cancellationToken)
+    {
+        Logger.LogTrace($"Connecting to named pipe {pipeName}");
+        Debug.Assert(_namedPipe is null);
+
+#if NET
+        _namedPipe = new NamedPipeClientStream(".", pipeName, PipeDirection.In, PipeOptions.CurrentUserOnly | PipeOptions.Asynchronous);
+#else
+        _namedPipe = new NamedPipeClientStream(".", pipeName, PipeDirection.In, PipeOptions.Asynchronous);
```

Comment:
For the .NET Framework case how are you validating the security of the named pipe? There is some code you can steal from roslyn to do this.

https://github.com/dotnet/roslyn/blob/main/src/Compilers/Shared/NamedPipeUtil.cs#L54

Reply:
Is that only for framework? Since this should only be in vs code we could probably make this only available in net if that helps simplify some of the requirements.

Reply:
That is only a requirement for framework. On .NET Core you just use `PipeOptions.CurrentUser`. Under the hood it is doing essentially that exact check.

Reply:
Can/should we move this implementation to rzls, so its .NET Core only? Certainly don't expect it to be used anywhere else, and there is already a facility to pass in a project info driver into the language server host.

Reply:
@ryzngard just checking that you saw my above comment? Might simplify a few things if all of this code exists exclusively in a .NET 8 project.

Reply:
Looking. It's just different from our _usual_ layering but probably can make it work without too much fuss

Reply:
The VS project info driver is in our VS layer, so whilst its different, its consistently different for this interface at least :)

Reply:
Yea, it's just a matter of swapping some things over into the `configureServices` call instead of passing directly.

Reply:
Next change is hopefully right. There's some ignoring of warnings that I don't know how to fix correctly but that's just lack of exposure.